setup.py

```diff
@@ -364,7 +364,7 @@ def parse_setuppy_commands():
 
 
 def setup_package():
-    src_path = os.path.dirname(os.path.abspath(sys.argv[0]))
+    src_path = os.path.dirname(os.path.abspath(__file__))
```

Comment:
When using `pyproject.yaml`, `setup.py` is now called via `exec(compile(code, __file__, 'exec'), locals())` in `python pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmp6g6mz9bq`. Since it is not called directly on the command line in a subprocess but via `exec`, `sys.argv[0]` is no longer `./setup.py`.

Reply:
yeah, just tracked down this too, but did not manage to quite get to the bottom of everything, maybe we can fix it tomorrow, or push towncrier to give an alternative toml file... (I commented on an issue there)

Reply:
This change looks harmless to me - functions in `setup.py` aren't public API, and any use-case involving symlinks to setup.py seems incredibly contrived